src/cmd/tools/zarf.go

```diff
@@ -225,6 +117,167 @@ func printComponentCredential(ctx context.Context, state *types.ZarfState, compo
 	}
 }
 
+// UpdateCredsOptions holds the command-line options for 'tools update-creds' sub-command.
+type UpdateCredsOptions struct{}
+
+// NewUpdateCredsCommand creates the `tools update-creds` sub-command.
+func NewUpdateCredsCommand(v *viper.Viper) *cobra.Command {
+	o := UpdateCredsOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "update-creds",
+		Short:   lang.CmdToolsUpdateCredsShort,
+		Long:    lang.CmdToolsUpdateCredsLong,
+		Example: lang.CmdToolsUpdateCredsExample,
+		Aliases: []string{"uc"},
+		Args:    cobra.MaximumNArgs(1),
+		RunE:    o.Run,
+	}
+
+	// Always require confirm flag (no viper)
+	cmd.Flags().BoolVar(&config.CommonOptions.Confirm, "confirm", false, lang.CmdToolsUpdateCredsConfirmFlag)
+
+	// Flags for using an external Git server
+	cmd.Flags().StringVar(&updateCredsInitOpts.GitServer.Address, "git-url", v.GetString(common.VInitGitURL), lang.CmdInitFlagGitURL)
+	cmd.Flags().StringVar(&updateCredsInitOpts.GitServer.PushUsername, "git-push-username", v.GetString(common.VInitGitPushUser), lang.CmdInitFlagGitPushUser)
+	cmd.Flags().StringVar(&updateCredsInitOpts.GitServer.PushPassword, "git-push-password", v.GetString(common.VInitGitPushPass), lang.CmdInitFlagGitPushPass)
```

Comment:
Yeah, I've noticed that change, but I didn't dig too deep into where it came from. I'll check if the order you're talking about has an effect. Honestly, I'd prefer to rewrite it such that the order shouldn't matter.

Reply:
> We don't want to remove `v.GetString(common.VInitGitPushUser)` from this line since a user may set `init.git.push_username` in their config.toml file. Instead we could move `rootCmd.AddCommand(tools.NewToolsCommand())` to be first in `NewZarfCommand()`.

Sounds like we should consider defaulting and config values, and how these interact. I think once I reach the point where I will want to decouple config values and flags it might get addressed.